demoproviders/barbican/java/org/openyolo/demoprovider/barbican/provider/CredentialPickerActivity.java

```diff
@@ -107,6 +107,13 @@ private void setResultAndFinish(CredentialRetrieveResult result) {
                     CredentialConverter.PROTO_TO_CREDENTIAL);
         } catch (IOException ex) {
             throw new IllegalStateException("Failed to decode credentials from intent", ex);
+        } catch (RuntimeException ex) {
```

Comment:
this is pretty ugly; I take it the potential cause is ProtoListUtil.readMessageList? Could we have that throw MalformedDataException directly?